ibm/service/kubernetes/resource_ibm_container_ingress_instance.go

```diff
@@ -0,0 +1,252 @@
+	return &iBMContainerIngressInstanceValidator
+}
+func resourceIBMContainerIngressInstanceCreate(d *schema.ResourceData, meta interface{}) error {
+	ingressClient, err := meta.(conns.ClientSession).VpcContainerAPI()
+	if err != nil {
+		return err
+	}
+
+	instanceCRN := d.Get("instance_crn").(string)
+	cluster := d.Get("cluster").(string)
+
+	params := v2.InstanceRegisterConfig{
+		CRN:     instanceCRN,
+		Cluster: cluster,
+	}
+
+	if v, ok := d.GetOk("is_default"); ok {
+		params.IsDefault = v.(bool)
+	}
+
+	if v, ok := d.GetOk("secret_group_id"); ok {
+		params.SecretGroupID = v.(string)
+	}
+
+	ingressAPI := ingressClient.Ingresses()
+	response, err := ingressAPI.RegisterIngressInstance(params)
+
+	if err != nil {
+		return err
+	}
+	d.SetId(fmt.Sprintf("%s/%s", cluster, response.Name))
+
+	return resourceIBMContainerIngressInstanceRead(d, meta)
+}
+
+func resourceIBMContainerIngressInstanceRead(d *schema.ResourceData, meta interface{}) error {
+	ingressClient, err := meta.(conns.ClientSession).VpcContainerAPI()
+	if err != nil {
+		return err
+	}
+	parts, err := flex.IdParts(d.Id())
+	if err != nil {
+		return err
+	}
+	clusterID := parts[0]
+	instanceName := parts[1]
+
+	ingressAPI := ingressClient.Ingresses()
+	ingressInstanceConfig, err := ingressAPI.GetIngressInstance(clusterID, instanceName)
+	if err != nil {
+		return err
+	}
+	d.Set("cluster", ingressInstanceConfig.Cluster)
+	d.Set("instance_name", ingressInstanceConfig.Name)
+	d.Set("instance_crn", ingressInstanceConfig.CRN)
+	d.Set("is_default", ingressInstanceConfig.IsDefault)
+	d.Set("secret_group_id", ingressInstanceConfig.SecretGroupID)
+	d.Set("secret_group_name", ingressInstanceConfig.SecretGroupName)
+	d.Set("instance_type", ingressInstanceConfig.Type)
+	d.Set("status", ingressInstanceConfig.Status)
+	d.Set("user_managed", ingressInstanceConfig.UserManaged)
+
+	return nil
+}
+
+func resourceIBMContainerIngressInstanceDelete(d *schema.ResourceData, meta interface{}) error {
+	ingressClient, err := meta.(conns.ClientSession).VpcContainerAPI()
+	if err != nil {
+		return err
+	}
+
+	ingressAPI := ingressClient.Ingresses()
+
+	parts, err := flex.IdParts(d.Id())
+	if err != nil {
+		return err
+	}
+	clusterID := parts[0]
+	instanceName := parts[1]
+
+	params := v2.InstanceDeleteConfig{
+		Cluster: clusterID,
+		Name:    instanceName,
+	}
+
+	err = ingressAPI.DeleteIngressInstance(params)
+	if err != nil {
+		return err
+	}
+
+	d.SetId("")
+	return nil
+}
+
+func resourceIBMContainerIngressInstanceUpdate(d *schema.ResourceData, meta interface{}) error {
+	ingressClient, err := meta.(conns.ClientSession).VpcContainerAPI()
+	if err != nil {
+		return err
+	}
+	parts, err := flex.IdParts(d.Id())
+	if err != nil {
+		return err
+	}
+	cluster := parts[0]
+	instanceName := parts[1]
+
+	params := v2.InstanceUpdateConfig{
+		Cluster: cluster,
+		Name:    instanceName,
+	}
+
+	if v, ok := d.GetOk("is_default"); ok {
```

Comment:
maybe check if these fields have changed with `d.HasChange` and only call the UpdateIngressInstance if so

Reply:
Do you mind going into what the difference between GetOK and HasChange is? The impression I'm getting from HasChange is that it's checking if the value has changed from before, whereas GetOK checks to see if a value has been set in general.